test/suite/test_timestamp14.py

```diff
@@ -0,0 +1,244 @@
+            lambda: self.conn.query_timestamp('get=all_committed'))
+
+        # Scenario 1: A single transaction with a commit timestamp, will
+        # result in the all_committed timestamp being set.
+        session1.begin_transaction()
+        cur1[1]=1
+        session1.commit_transaction('commit_timestamp=1')
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "1")
+
+        # Scenario 2: A transaction begins and specifies that it intends
+        # to commit at timestamp 2, a second transaction begins and commits
+        # at timestamp 3.
+        session1.begin_transaction()
+        session1.timestamp_transaction('commit_timestamp=2')
+
+        session2.begin_transaction()
+        cur2 = session2.open_cursor(all_committed_uri)
+        cur2[2] = 2
+        session2.commit_transaction('commit_timestamp=3')
+
+        # As the original transaction is still running the all_committed
+        # timestamp is being held at 1.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "1")
+        cur1[1] = 2
+        session1.commit_transaction()
+
+        # Now that the original transaction has finished the all_committed
+        # timestamp has moved to 3, skipping 2 as there is a commit with
+        # a greater timestamp already existing.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "3")
+
+        # Senario 3: Commit with an all committed of 5 and then begin a
+        # transaction intending to commit at 4, the all_committed timestamp
+        # should move back.
+        session1.begin_transaction()
+        cur1[1] = 3
+        session1.commit_transaction('commit_timestamp=5')
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "5")
+
+        session1.begin_transaction()
+        # All committed will now move back to 3 as it is the point at which
+        # all transactions up to that point have committed.
+        session1.timestamp_transaction('commit_timestamp=4')
+
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "3")
+
+        session1.commit_transaction()
+
+        # Now that the transaction at timestamp 4 has completed the
+        # all committed timestamp is back at 5.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "5")
+
+        # Scenario 4: Holding a transaction open without a commit timestamp
+        # Will not affect the all_committed timestamp.
+        session1.begin_transaction()
+        session2.begin_transaction()
+        cur2[2] = 2
+        session2.commit_transaction('commit_timestamp=6')
+
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "6")
+        cur1[1] = 2
+        session1.commit_transaction()
+
+    def test_oldest_reader(self):
```

Comment:
can add some scenario's to use set_timestamp API for oldest timestamp and check the behavior.